### DISCUSSION TITLE
Integration thread - locked

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#891] [Ideas]
Hey! 

If Budibase does not support an integration you need, please detail it in the comments below. Please review comments before submitting to avoid duplication. To add weight to suggested integrations, you can upvote them with a 👍 .

Thank you!

**Comment by Wulfheart** [comment]
Looked at this thread too late. Therefore I created #906.
From my experience developing internal tools two things are needed:
MySQL/MariaDB and Microsoft SQL Server support as these belong to the most used database systems and the possibility to connect to a local mailserver via IMAP or SMTP.

**Reply by fabianmossberg** [reply]
MySQL please 🙏

**Reply by joebudi** [reply]
MySQL is live. Accessible via data sources

**Reply by fabianmossberg** [reply]
I was lazy, just looked at the website. But now I'm building stuff with this. I love it!

**Reply by jekroll555** [reply]
YES...Microsoft SQL Server is a must have for me

**Reply by sladerose** [reply]
A map feature, like SAP GeoMaps! This is a must have for our workflows.

**Comment by joebudi** [comment]
Github integration! 

I would love to have issues from Github feed into my BB app. I would also like to create a dashboard with stats such as PRs, Issues, Forks, Commits, and Stars - possibly Discussions too!

**Comment by uelsimon** [comment]
As a non-technical founder,  being able to use Airtable might be nice.  (RestAPI and GraphQL support also)

**Reply by joebudi** [reply]
Hey @uelsimon 

Thank you for your suggestion . The Airtable integration will be available in the next week or two, along with others (Postgres, Couch, Mongo). 

I am interested to know what you expect of such an integration? What do you expect the UX to look like? @shogunpurple might have additional questions / info regarding the UX.

Thanks
Joe

**Reply by joebudi** [reply]
The Airtable integration is live. The RestAPI is also live.

**Comment by t2the2ndpower** [comment]
Salesforce.com integration would be way cool!

**Reply by joebudi** [reply]
This is not on our immediate roadmap but we understand it is essential. I feel this particular request will be very popular.

**Comment by Bogey** [comment]
Google’s Firebase Auth, Firebase Firestore and Firebase Realtime.

**Comment by SOVLOOKUP** [comment]
[arangodb](https://github.com/arangodb/arangodb)!
SQL and document databases can no longer meet the needs(graph data,geo data),arangodb integration might be cool.

**Reply by joebudi** [reply]
I've never heard of this DB and it looks interesting. This is not on our roadmap, but integrating external DBs will soon be a straightforward task. We're open-source, and if you have the time, adding arangodb as a Budibase external DB would be an incredible contribution. We're happy to help you with this, and provide some SWAG too :-) if you are interested.

**Reply by SOVLOOKUP** [reply]
I'm glad to do this. Just after my more exploration in budibase. Is there any develop instructions?
In addition, is there any plan to make the builder support multiple language?

**Reply by joebudi** [reply]
@shogunpurple is writing the docs over the next few days. He will be happy to work with you to achieve this. He'll also collect your details so we can send you some SWAG.

**Reply by shogunpurple** [reply]
Hi @SOVLOOKUP, here are the docs @joebudi mentioned for writing external data connectors. Feel free to jump into discord and ask more questions if you like. 

https://docs.budibase.com/contributions/contribute-data-connectors

**Comment by jibize** [comment]
[FaunaDB](https://fauna.com/) would be a nice addition to the list.

**Comment by LegacyDebaditya** [comment]
adding Github and Gitlab would create a deep effect

**Reply by mjashanks** [reply]
Thanks! What would you expect to do with a Github / Gitlab integration?

**Comment by OccultSlolem** [comment]
I would love to have an integration with Firebase. I use Realtime Database to store user data, Firebase Auth for authentication, and Storage for photos.

**Reply by joebudi** [reply]
This has been requested a few times so I am going to chat to the team about getting it into the next sprint.

**Reply by OccultSlolem** [reply]
Hell yeahhhhhh

**Comment by projx** [comment]
I'd like to request integration with Splunk search please.

**Reply by joebudi** [reply]
We played around with integrating Splunk a few weeks back. It's great. I think there's a possibility we will integrate with it in the next few months.

**Comment by cptr13** [comment]
Habitica.... Please. They have a good api.

**Reply by joebudi** [reply]
Oh. This is new to me. Will have a look. Thanks.

**Comment by bsieber-mozilla** [comment]
An (Open Source) Tracking System would be a worthwhile integration. 

Example: I'd like to track my work in a tracking system, and when bugs/issues are in a particular state or undergo a change: send an email, put (time-tracking) data in db, send notes to wiki,...

Here's a list of potential options to look into...

- https://github.com/bugzilla/bugzilla
- https://github.com/opf/openproject 
- https://github.com/Leantime/leantime
- https://github.com/Enalean/tuleap
- https://github.com/taigaio 
- https://github.com/RestyaPlatform/board  
- https://github.com/icescrum/iceScrum
- https://github.com/digitalfondue/lavagna

Appreciate all your hard work in making a beautiful product, while also assisting the Open Source and SelfHosting communities. (Found BudiBase though /r/selfhosted.) 

(Also, if this doesn't align with Budibase's vision that's totally understandable!)

_I'll definitely be keeping an eye on Budibase and tooting a horn for y'all._

**Reply by shogunpurple** [reply]
This is awesome - thank you for taking the time. We will be implementing a REST API integration very soon - allowing people to effectively connect to any HTTP API they want. That said, much tighter integration with the tools you mentioned could be very interesting for the use cases you stated.

**Reply by bsieber-mozilla** [reply]
_Thanks for reading and responding!_ 

That REST API integration sounds like a core feature add; that's great news! 

Keep up the awesome work, thanks again! 😄

**Reply by Jared-Dev** [reply]
Is there any update on allowing for a REST API datasource? Just found Budibase and this is one feature I'd love to see!

**Reply by joebudi** [reply]
Hey @Jared-Dev . We are planning a release tomorrow which will allow you to use a Rest API as a data source :-)

**Comment by icecool69oh** [comment]
It would be awesome to see an integration for Firebird SQL.

**Reply by joebudi** [reply]
Oh nice. Thanks for reaching out.

**Comment by thiagoferolla** [comment]
Please, Firebase Realtime Database, Firestore and Firebase Auth

**Reply by joebudi** [reply]
Firebase is a very popular choice. We will look into it over the next few weeks.

**Comment by AlexRMU** [comment]
Node.js and npm modules

**Reply by mjashanks** [reply]
Hey. Thanks for the suggestion.

At some point, in the near future we hope to allow you to write javascript to add custom backend integrations, and frontend components.

I.e. without having to wait for it to be contributed to the core codebase.

Is this the kind of thing that you mean?

**Reply by AlexRMU** [reply]
Yes, approximately

**Comment by indopospay** [comment]
Please add Airtable, Supabase.io (as Firebase alternative) and Tallyform

**Reply by mjashanks** [reply]
Airtable is already here! It's a data source, you can alongside tables.

Supabase is a great shout. Fits really well with the open source, self hosting story.

**Reply by indopospay** [reply]
Whoaa.. fast response as always. 

And may I know is payment-gateway already exist ? 
Thank you

**Comment by darccio** [comment]
I think that Amazon SES would be a great addition. Also LDAP for authentication but I know it's harder to integrate.

**Reply by joebudi** [reply]
Improved Auth is a popular request and something we will focus heavily on over the next few weeks. We will consider services such as passportjs but more thought is required.  Any guidance is appreciated.

**Comment by AlexRMU** [comment]
[Svelte](https://svelte.dev/) integration

**Reply by mjashanks** [reply]
Hey Alex. We plan to allow you to write custom components in Svelte, soon, but not sure exactly when :)

**Comment by moming2k** [comment]
Google Sheet integration would be great

**Reply by indopospay** [reply]
vote for this

**Reply by joebudi** [reply]
This will come in the next few weeks

**Reply by Caterina1994** [reply]
Hey @joebudi  any update on google sheets :) ?

**Reply by joebudi** [reply]
Hey @Caterina1994 . It's on the roadmap, but it won't be in the next release (Thurs). We are having a strategy meeting next week, so I will have more information then. I expect it will be done in the next month or two.

**Reply by Caterina1994** [reply]
Thank you @joebudi ! I wanted to know so I can find my way into another solution as my Database is in Big query and i need to execute actions when information is updated in tables that affect other tables.. so I thought as google sheets of a solution, but will make my way around this. thank you!

**Reply by joebudi** [reply]
Ahhh. We support other data sources, such as airtable, that might help.

**Comment by michaelmior** [comment]
Would love to see Cassandra integrated :)

**Reply by joebudi** [reply]
Oh. This is an interesting suggestion. It's one we haven't considered but would be happy to implement if the demand is there.

**Comment by I-Connect** [comment]
An MQTT connector would be nice:

- Ability to subscribe to topics
- Storing MQTT messages in the database
- Option to enrich the MQTT message record that will be stored with additional data depending on topic or message
- (Maybe also add this as an input AND output for the automations...)

**Comment by ZiffyZed** [comment]
LDAP integration would be great!
It is very useful to import list of your users from project to project.

**Reply by joebudi** [reply]
This is a popular request and one we will need to integrate over the upcoming months.

**Comment by OccultSlolem** [comment]
Adding support for social media auth (especially Google, Apple, GitHub, and Twitter) would be pretty pog

**Reply by joebudi** [reply]
Hey. We now have Google Auth with the latest release:
https://github.com/Budibase/budibase/discussions/1557

**Comment by Caterina1994** [comment]
Hi, BIG QUERY PLEASE!!!

**Reply by joebudi** [reply]
This is a great shout! I am going to reference this in a new issue. May I ask what your profession is? I am trying to tie professions to popular integrations :-)

**Reply by Caterina1994** [reply]
Hey @joebudi  thanks for the response. Im a business intelligence Analyst working for a company and we have our data in Big query.
I would like to create an internal tool for this company.
Thanks!

**Reply by joebudi** [reply]
I appreciate your quick response. This information is very valuable to us. We are finding more and more analysts signing up to Budibase and I want to make sure we are providing the right components / functionality to help with development.
I've raised this integration request as an issue as it's been a very popular one.

**Reply by Caterina1994** [reply]
Great @joebudi  thanks!  In Meantime i guess i will have to use google sheets as my database...
How can i find out if this will actually become a feature?

**Reply by joebudi** [reply]
I'll keep you informed with this thread. Google sheets is another integration coming - you've basically touched on the next two integrations :-)

**Reply by Caterina1994** [reply]
Super!! thanks so much :).. by the way .. where do i comunicate to get support with hosting?

**Reply by joebudi** [reply]
If it is a problem with the platform/docs, simply create a new issue and outline your problem:
https://github.com/Budibase/budibase/issues/new/choose

If you are simply looking for advice, raise a discussion with the Q&A tag:
https://github.com/Budibase/budibase/discussions/new

I hope that helps :-)

**Comment by arladmin** [comment]
I think, adding n8n.io integration would probably facilitate most of the use-cases mentioned above!

**Reply by joebudi** [reply]
Great shout. n8n has always been on the back of minds. We just need to find the time to implement it! We do have integromat and Zapier.

**Reply by berserkwarwolf** [reply]
@joebudi Can we put a bounty on this ? My company would be willing to invest.

**Reply by joebudi** [reply]
This will get done, sooner rather than later. I'll try to get it in for the next release.

**Comment by FloMiLe** [comment]
Sending mails via SMTP would be nice.

**Reply by joebudi** [reply]
This is on the roadmap and will be done within the next couple of months.

**Reply by joebudi** [reply]
This is possible with the new release :-)

https://github.com/Budibase/budibase/discussions/1557

**Comment by allthesignals** [comment]
A generic read-only JSON adapter would be cool — something like reading simple array of records. No write functionality needed.

**Reply by joebudi** [reply]
Interesting. We will look into this.

**Comment by acidlake** [comment]
GraphQL would be nice and Rest API

**Reply by joebudi** [reply]
Rest API will be released tomorrow! GraphQL is a popular request, and I am personally a fan. It's not on our roadmap but I feel it will come in the next month or two.

**Reply by acidlake** [reply]
Great!! waiting to test Rest API, thanks a lot for the great work

**Reply by joebudi** [reply]
The rest api is live now

**Comment by SOVLOOKUP** [comment]
[Parse Platform](https://parseplatform.org/) Another opensource Firebase alternative.

**Reply by joebudi** [reply]
Parse is such an interesting platform. We don't have plans to support it but it is something we will consider.

**Comment by hansaliyad1** [comment]
Microsoft SQL as a data source

**Comment by wiben-sound** [comment]
google oauth to replace login

**Reply by joebudi** [reply]
Oauth will come. We're meeting on Tuesday to discuss strategy and our roadmap. I'll have more info then.

**Comment by FaustinM** [comment]
A redis integration :) !
Thanks you

**Reply by joebudi** [reply]
Ohhh. Interesting. Out of curiosity, what is the use case?

**Reply by FaustinM** [reply]
We have a cache with all player inventory of our game and sometime we need to manage this cache by example.

**Comment by changwuf31** [comment]
ClickHouse please, it's very fast

**Reply by joebudi** [reply]
Clickhouse is awesome. We're in heavy development at the moment, so we won't get to this for a while. If you need it quicker, you could add it yourself:
https://docs.budibase.com/contributions/contribute-data-connectors

If you do decide to add it, let us know if you run into any issues and we'd be happy to help.

**Comment by muras** [comment]
Please add integration with google docs (excel) to dynamically load data from sheets (now we need to upload CSV file manually to refresh data)

**Reply by joebudi** [reply]
This is coming. A Google sheets integration is on the roadmap.

**Comment by Grumpypjs08** [comment]
Some sort of file store with private/roll download links

**Reply by joebudi** [reply]
Interesting. Could you expand a little so I can bring this to our roadmap meeting. It would be good to understand the use case behind the requirement.

**Comment by vinith39** [comment]
Google map integration ? I plan to track my users live.

**Comment by hshifaz** [comment]
Hi Guys, great work. I just recently installed self hosting and everything seems very easy to follow. 

I do however like to know if it is possible to integrate with MS AD for user authentication.  Like for example using LDAP.

**Reply by joebudi** [reply]
Hey. Not right now, but it is a very popular request. I don't have a timeline for this, but it is a priority.

**Reply by hshifaz** [reply]
Great, can't wait.
So, mean time is there a work around. I'm kinda working on a mini project, with rapid app development and extreme programming concepts in mind, to provide simple tools to solve tedious tasks. If I can use AD auth it'll be easier for me to onboard others in my team because we have many systems using AD for user authentication.

**Reply by shogunpurple** [reply]
@hshifaz there is no workaround for AD authentication. We will be implementing this in future though. Thanks for the suggestion

**Reply by hshifaz** [reply]
Great, thanx for the prompt response. I like this team.

**Comment by joebudi** [comment]
I am locking this conversation as ideas/feature requests should be separate discussions rather than documented within a thread. If you would like to suggest a new integration, please create a new discussion under the idea category - thanks!
